src/middleware/createRouterMiddleware.js

```diff
@@ -442,9 +476,20 @@ const createRouterMiddleware = ({
         return undefined;
       },
 
-      [ActionTypes.CANCEL_ROUTE]({action, next}) {
+      [ActionTypes.CANCEL_ROUTE]({ action, next }) {
         inFlightNext = null;
         return next(action);
+      },
+
+      [ActionTypes.RESET_LOCATION]({ action, next }) {
+        const newNextAction = {
+          ...action,
+          payload: {
+            location: parseUrl(action.payload.href, baseUrl)
```

Comment:
Maybe we could  default to the `browserUrl()` here for convenience?

```
  location: parseUrl(action.payload.href || browserUrl(), baseUrl)
```

Reply:
yeah that makes sense to me